integration/tests/epochs/epoch_test.go

```diff
@@ -136,31 +138,52 @@ func (s *Suite) TestEpochJoin() {
 
 	env := utils.LocalnetEnv()
 
+	role := flow.RoleAccess
 	// stake a new node
-	info := s.StakeNode(ctx, env, flow.RoleConsensus)
-
-	// get node info from staking table
-	nodeInfoCDC := s.ExecuteGetNodeInfoScript(ctx, env, info.NodeID)
-	nodeInfo, ok := nodeInfoCDC.(cadence.Struct)
-	require.True(s.T(), ok)
+	info := s.StakeNode(ctx, env, role)
+	testContainerName := fmt.Sprintf("epochs-test-join-%s-%s", info.Role, info.NodeID)
 
-	// make sure node info we generated matches what we get from the flow staking table
-	nodeIDFromState := string(nodeInfo.Fields[0].(cadence.String))
-	require.Equal(s.T(), info.NodeID.String(), nodeIDFromState, "expected node ID generated in test to equal node ID from staking table ")
+	// check if node is in proposed table
+	proposedTable := s.ExecuteGetProposedTableScript(ctx, env, info.NodeID)
+	assert.Contains(s.T(), proposedTable, info.NodeID, "expected new node to be in proposed table")
 
 	result := s.SetApprovedNodesScript(ctx, env, append(s.net.Identities().NodeIDs(), info.NodeID)...)
 	require.NoError(s.T(), result.Error)
 
-	// get new approved nodes list and make sure new node was added correctly
+	// ensure node ID in approved list
 	approvedNodes := s.ExecuteReadApprovedNodesScript(ctx, env)
+	assert.Contains(s.T(), approvedNodes, info.NodeID)
+
+	nodeConfig := testnet.NewNodeConfig(role, testnet.WithID(info.NodeID))
+	testContainerConfig := testnet.NewContainerConfig(testContainerName, nodeConfig, info.NetworkingKey, info.StakingKey)
+	err := testContainerConfig.WriteKeyFiles(s.net.BootstrapDir, flow.Localnet, info.MachineAccountAddress, encodable.MachineAccountPrivKey{PrivateKey: info.MachineAccountKey}, role)
+	require.NoError(s.T(), err)
+
+	// download root snapshot from access node, wait until we are in the epoch setup phase
+	var snapshot *inmem.Snapshot
+	for {
+		snapshot, err = s.client.GetLatestProtocolSnapshot(ctx)
+		require.NoError(s.T(), err)
 
-	found := false
-	for _, val := range approvedNodes.(cadence.Array).Values {
-		if string(val.(cadence.String)) == info.NodeID.String() {
-			found = true
+		currentPhase, err := snapshot.Phase()
+		require.NoError(s.T(), err)
+		_, err = snapshot.Identity(info.NodeID)
+
+		if currentPhase == flow.EpochPhaseSetup && err == nil {
+			break
 		}
 	}
 
-	require.True(s.T(), found, "node id for new node not found in approved list after setting the approved list")
-	s.net.StopContainers()
+	// write updated root snapshot
+	s.net.WriteRootSnapshot(snapshot)
```

Comment:
This is writing the root snapshot for the whole network, which I don't think we want here. We're emulating one node joining the existing network, so only that one node would have the later snapshot.